.nix/coq-overlays/coq-full/default.nix

```diff
@@ -0,0 +1,151 @@
+{ lib, stdenv, hostname, writeText
+, ocamlPackages, dune_3, ocamlformat
+, python3, time, glib, wrapGAppsHook, pkg-config
+, antlr4, ncurses, rsync, which, jq, curl, gitFull, gnupg, graphviz, gnome
+, buildIde ? true
+, buildDoc ? true
+, doInstallCheck ? true
+, shell ? true
+, coq-version ? "8.17-git"
+, ...
+}:
+
+with lib;
+
+stdenv.mkDerivation rec {
+
+  pname = "coq-full";
+  version = "dev";
+
+  buildInputs = [
+    hostname
+    python3
+    # coq-makefile timing tools
+    time
+    dune_3
+  ]
+  ++ optionals buildIde [
+    ocamlPackages.lablgtk3-sourceview3
+    glib
+    gnome.adwaita-icon-theme
+    wrapGAppsHook
+  ]
+  ++ optionals buildDoc [
+    # Sphinx doc dependencies
+    pkg-config (python3.withPackages
+      (ps: [ ps.sphinx ps.sphinx_rtd_theme ps.pexpect ps.beautifulsoup4
+             ps.antlr4-python3-runtime ps.sphinxcontrib-bibtex ]))
+    antlr4
+    ocamlPackages.odoc
+  ]
+  ++ optionals doInstallCheck [
+    # Test-suite dependencies
+    ocamlPackages.ounit
+    rsync
+    which
+  ]
+  ++ optionals shell (
+    [ # Dependencies of the merging script
+      jq
+      curl
+      gitFull
+      gnupg
+    ]
+    ++ (with ocamlPackages; [
+      # Dev tools
+      ocaml-lsp
+      merlin
+      ocp-indent
+      ocp-index
+      utop
+      ocamlformat
+      ])
+    ++ [
+      # Useful for STM debugging
+      graphviz
+    ]
+  );
+
+  # OCaml and findlib are needed so that native_compute works
+  # This follows a similar change in the nixpkgs repo (cf. NixOS/nixpkgs#101058)
+  # ocamlfind looks for zarith when building plugins
+  # This follows a similar change in the nixpkgs repo (cf. NixOS/nixpkgs#94230)
+  propagatedBuildInputs = with ocamlPackages; [ ocaml findlib zarith ];
+  ocamlBuildInputs = propagatedBuildInputs;
+
+  propagatedUserEnvPkgs = with ocamlPackages; [ ocaml findlib ];
+
+  src =
+    with builtins; filterSource
+      (path: _:
+        !elem (baseNameOf path) [".git" "result" "bin" "_build" "_build_ci" "_build_vo" ".nix"]) ../../..;
+
+  preConfigure = ''
+    patchShebangs dev/tools/ doc/stdlib
+  '';
+
+  prefixKey = "-prefix ";
+
+  enableParallelBuilding = true;
+
+  buildFlags = [ "world" ];
+
+  # TODO, building of documentation package when not in dev mode
+  # https://github.com/coq/coq/issues/16198
+  # buildFlags = [ "world" ] ++ optional buildDoc "refman-html";
+
+  # From https://github.com/NixOS/nixpkgs/blob/master/pkgs/build-support/ocaml/dune.nix
+  installPhase = ''
+    runHook preInstall
+    dune install --prefix $out --libdir $OCAMLFIND_DESTDIR coq-core coq-stdlib coq coqide-server coqide
+    runHook postInstall
+  '';
+
+  # installTargets =
```

Comment:
What to do with this?

Reply:
Note that this file is almost identical to what is currently in `default.nix` and the commit history traced that. See 57e4fef for the differences. This is part of the legacy of #15560. IIUC this is related to the doc build being disabled. That being said, since I see a comment "TODO, building of documentation package when not in dev mode", this hints that a solution is to build in dev mode. In this case, we would have to skip the install phase entirely, but we would be able to bring back the documentation build...